src/SFA.DAS.AssessorService.Application.Api.Client/Azure/AzureApiClient.cs

```diff
@@ -0,0 +1,243 @@
+
+            using (var httpRequest = new HttpRequestMessage(HttpMethod.Put, $"/subscriptions/{subscriptionId}?api-version=2017-03-01"))
+            {     
+                return await PostPutRequestWithResponse<CreateAzureUserSubscriptionRequest, AzureSubscription>(httpRequest, request);
+            }
+        }
+
+        public async Task EnableUser(string userId)
+        {
+            var user = await GetUserDetails(userId);
+
+            if (user != null)
+            {
+                var request = new EnableDisableAzureUserRequest { State = "active" };
+                using (var httpRequest = new HttpRequestMessage(new HttpMethod("PATCH"), $"/users/{userId}?api-version=2017-03-01"))
+                {
+                    httpRequest.Headers.Add("If-Match", "*");
+                    await PostPutRequest(httpRequest, request);
+                }
+            }
+            else
+            {
+                throw new Exceptions.EntityNotFoundException($"Could not find User");
+            }
+        }
+
+        public async Task DisableUser(string userId)
+        {
+            var user = await GetUserDetails(userId);
+
+            if (user != null)
+            {
+                var request = new EnableDisableAzureUserRequest { State = "blocked" };
+                using (var httpRequest = new HttpRequestMessage(new HttpMethod("PATCH"), $"/users/{userId}?api-version=2017-03-01"))
+                {
+                    httpRequest.Headers.Add("If-Match", "*");
+                    await PostPutRequest(httpRequest, request);
+                }
+            }
+            else
+            {
+                throw new Exceptions.EntityNotFoundException($"Could not find User");
+            }
+        }
+
+        public async Task DeleteUser(string userId)
+        {
+            var user = await GetUserDetails(userId);
+
+            if (user != null)
+            {
+                using (var httpRequest = new HttpRequestMessage(HttpMethod.Delete, $"/users/{userId}?api-version=2017-03-01&deleteSubscriptions=true"))
+                {
+                    httpRequest.Headers.Add("If-Match", "*");
+                    await Delete(httpRequest);
+                }
+
+                //var organisation = await _organisationsApiClient.Get(user.Note);
+                //await _organisationsApiClient.Update(new UpdateOrganisationRequest
+                //{
+                //    EndPointAssessorName = organisation.EndPointAssessorName,
+                //    EndPointAssessorOrganisationId = organisation.EndPointAssessorOrganisationId,
+                //    EndPointAssessorUkprn = organisation.EndPointAssessorUkprn,
+                //    ApiEnabled = false,
+                //    ApiUser = null
+                //});
+            }
+
+        }
+
+        public async Task<IEnumerable<AzureProduct>> ListProducts()
```

Comment:
Not sure if we need localisation with resource strings? for this project.

Reply:
Localisation hasn't been spec'ed as of yet. @davidgouge could you let me know if you want this.